<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="Tade Souaiaia" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>Input Data - BridgePRS</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link href="../css/extra.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Input Data";
        var mkdocs_page_input_path = "guide_input.md";
        var mkdocs_page_url = null;
      </script>
    
    <script src="../js/jquery-3.6.0.min.js" defer></script>
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]--> 
      <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

        ga('create', 'UA-113286405-1', 'auto');
        ga('send', 'pageview');
      </script>
    
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href=".." class="icon icon-home"> BridgePRS
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Quick Start Tutorial</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../quikstart_prep/">Before Running</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../quikstart_demo/">Toy-Data Demo</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../quikstart_result/">After Running</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../quikstart_challenges/">User Challenges</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Guide</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../guide_background/">Background</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_requirements/">Requirements</a>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" href="./">Input Data</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#ld-reference-data-1000-genomes">LD Reference Data: 1000 Genomes</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#trait-data-gwas-sumstats-and-phenotype-data">Trait Data: GWAS Sumstats and Phenotype Data</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#the-configuration-file">The Configuration File</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#file-specifications">File Specifications</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#1-sumstats-data">1) Sumstats Data</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#2-phenotype-files">2) Phenotype Files</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#3-qc-snp-list">3) QC SNP List</a>
    </li>
        </ul>
    </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_subprograms/">Subprograms</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_output/">Output Data</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_args/">All Arguments</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_customization/">Customization</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_shell/">Using Bash</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Misc</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../misc_faq/">Frequently Asked Questions</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../mac_specific/">Mac Specific</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../misc_log/">Update Log</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">BridgePRS</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" alt="Docs"></a> &raquo;</li>
          <li>Guide &raquo;</li>
      <li>Input Data</li>
    <li class="wy-breadcrumbs-aside">
        <a href="https://github.com/tadesouaiaia/BridgePRS-Guide/edit/master/docs/guide_input.md"
          class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <p><img alt="Screenshot" src="../img/slim/guide_logo3.png" /> </p>
<h1 id="input-data">Input Data</h1>
<h2 id="ld-reference-data-1000-genomes">LD Reference Data: 1000 Genomes</h2>
<p>BridgePRS estimates SNP-specific LD in the clumping step from an population target data in PLINK binary format.  By default, the main 
BridgePRS includes a sample LD reference panel (<strong>BRIDGEDIR/data/1000G_sample</strong>) for population super-groups <strong>AFR</strong> and <strong>EUR</strong> that is 
suitable for the <a href="../quikstart_prep/">quick start tutorial</a>. </p>
<p>A folder containing the full 1000 Genomes LD reference panel for populations <strong>AFR</strong>, <strong>EUR</strong>, <strong>EAS</strong>, <strong>SAS</strong>, and <strong>AMR</strong> can be downloaded 
<a href="https://github.com/clivehoggart/BridgePRS/archive/refs/heads/main.zip">here</a> and unzipped into <strong>BRIDGEDIR/data/1000G_ref</strong> to enable full usage 
of BridgePRS. </p>
<p>To run BridgePRS using a custom LD reference please see <a href="../guide_customization/">customization</a>. </p>
<h2 id="trait-data-gwas-sumstats-and-phenotype-data">Trait Data: GWAS Sumstats and Phenotype Data</h2>
<p><strong>Target Data</strong> </p>
<p>For the target population (where the trait is being estimated) BridgePRS requires GWAS summary stats
and two phenotype files (for test and validation): </p>
<table>
<thead>
<tr>
<th align="center">Name</th>
<th align="center">Command Line flag</th>
<th align="center">Required</th>
<th align="center">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">Sumstats</td>
<td align="center">--sumstats_prefix</td>
<td align="center">Yes</td>
<td align="center">GWAS Summary Statsd</td>
</tr>
<tr>
<td align="center">Phenotypes</td>
<td align="center">--phenotype_files</td>
<td align="center">Target Pop Only</td>
<td align="center">Test And Validation Phenotype Files</td>
</tr>
<tr>
<td align="center">Snp List</td>
<td align="center">--snp_file</td>
<td align="center">No</td>
<td align="center">List QCed SNP ids</td>
</tr>
</tbody>
</table>
<div class="admonition tip">
<p class="admonition-title">Creating a Configuration File</p>
<p>The following command will validate the input data supplied on the command line 
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span>./bridgePRS check data -o out --pop AFR 
                       --sumstats_prefix data/pop_africa/sumstats/afr.chr
                       --phenotype_files data/pop_africa/phenotypes/afr_test.dat data/pop_africa/phenotypes/afr_validation.dat
</pre></div>
And produce a configuration file for an african population that mirrors the supplied configuration file: </p>
<div class="tabbed-set" data-tabs="1:1"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><label for="__tabbed_1_1">$ cat data/afr_cont.config</label><div class="tabbed-content"></div>
</div>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span> <span style="color: #666666">.</span><span style="color: #008000; font-weight: bold">...</span>
 POP<span style="color: #666666">=</span>AFR
 SUMSTATS_PREFIX<span style="color: #666666">=</span>pop_africa<span style="color: #666666">/</span>sumstats<span style="color: #666666">/</span>afr.chr
 PHENOTYPE_FILES<span style="color: #666666">=</span>pop_africa<span style="color: #666666">/</span>phenotypes<span style="color: #666666">/</span>afr_test.dat,pop_africa<span style="color: #666666">/</span>phenotypes<span style="color: #666666">/</span>afr_validation.dat
 SNP_FILE<span style="color: #666666">=</span>pop_africa<span style="color: #666666">/</span>snps.txt
</pre></div>
</div>
<p><strong>Model Population</strong> </p>
<p>For the model population (used to produce prior distributions) BridgePRS requries GWAS summary stats 
and one phenotype file (if available).  If it is not available then the test file 
from the target population can be substituted, as shown in the two configuration files: </p>
<div class="tabbed-set" data-tabs="2:2"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><label for="__tabbed_2_1">With Phenotype Data</label><div class="tabbed-content">
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span>POP<span style="color: #666666">=</span>EUR 
SUMSTATS_PREFIX<span style="color: #666666">=</span>pop_europe<span style="color: #666666">/</span>sumstats<span style="color: #666666">/</span>eur.chr
PHENOTYPE_FILES<span style="color: #666666">=</span>pop_europe<span style="color: #666666">/</span>phenotypes<span style="color: #666666">/</span>eur_test.dat
SNP_FILE<span style="color: #666666">=</span>pop_europe<span style="color: #666666">/</span>snps.txt
</pre></div>
</div>
<input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><label for="__tabbed_2_2">Without Phenotype Data</label><div class="tabbed-content">
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span>POP<span style="color: #666666">=</span>EUR
SUMSTATS_PREFIX<span style="color: #666666">=</span>pop_europe<span style="color: #666666">/</span>sumstats<span style="color: #666666">/</span>eur.chr
PHENOTYPE_FILES<span style="color: #666666">=</span>pop_africa<span style="color: #666666">/</span>phenotypes<span style="color: #666666">/</span>afr_test.dat
SNP_FILE<span style="color: #666666">=</span>pop_europe<span style="color: #666666">/</span>snps.txt
</pre></div>
</div>
</div>
<h2 id="the-configuration-file">The Configuration File</h2>
<p>For ease of use bridgePRS uses configuration files. </p>
<h2 id="file-specifications">File Specifications</h2>
<h3 id="1-sumstats-data">1) Sumstats Data</h3>
<p>GWAS summary statistics can be provided using a prefix to multiple chromosome separated files with the <code>--sumstats_prefix</code> flag or 
with using a single sumstats file with the <code>--sumstats_file</code> flag.  If a prefix is given, then a suffix can be provided as well <code>--sumstats_suffix</code>. 
GWAS summary statistics must be provided as a whitespace delimited file containing association analysis results for SNPs on the base phenotype.
BridgePRS has no problem reading in a gzipped base file (need to have a <strong>.gz</strong> suffix) or splitting the file by chromosome if necessary.<br />
An example of a sumstats file with default column headers is shown: </p>
<table>
<thead>
<tr>
<th align="center">#CHROM</th>
<th align="center">POS</th>
<th align="center">ID</th>
<th align="center">REF</th>
<th align="center">ALT</th>
<th align="center">A1</th>
<th align="center">A1_FREQ</th>
<th align="center">TEST</th>
<th align="center">OBS_CT</th>
<th align="center">BETA</th>
<th align="center">SE</th>
<th align="center">T_STAT</th>
<th align="center">P</th>
<th align="center">ERRCODE</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">1</td>
<td align="center">754105</td>
<td align="center">rs12184325</td>
<td align="center">T</td>
<td align="center">C</td>
<td align="center">G</td>
<td align="center">0.0257573</td>
<td align="center">ADD</td>
<td align="center">4853</td>
<td align="center">0.820864</td>
<td align="center">0.413692</td>
<td align="center">1.98424</td>
<td align="center">0.0472871</td>
<td align="center">.</td>
</tr>
<tr>
<td align="center">1</td>
<td align="center">840753</td>
<td align="center">rs4970382</td>
<td align="center">C</td>
<td align="center">T</td>
<td align="center">G</td>
<td align="center">0.483495</td>
<td align="center">ADD</td>
<td align="center">4847</td>
<td align="center">0.0011142</td>
<td align="center">0.128347</td>
<td align="center">0.00868116</td>
<td align="center">0.993074</td>
<td align="center">.</td>
</tr>
<tr>
<td align="center">1</td>
<td align="center">958905</td>
<td align="center">rs2710890</td>
<td align="center">G</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">0.424387</td>
<td align="center">ADD</td>
<td align="center">4814</td>
<td align="center">0.108094</td>
<td align="center">0.132225</td>
<td align="center">0.817497</td>
<td align="center">0.413687</td>
<td align="center">.</td>
</tr>
</tbody>
</table>
<p>Column headers can be specified on the command line or in a configuration file using <code>--ssf</code> flags: 
(<code>--ssf-chr</code>,<code>--ssf-p</code>, <code>--ssf-snpid</code>, <code>--ssf-se</code>, <code>--ssf-ss</code>, <code>--ssf-beta</code>, <code>--ssf-ref</code>, <code>--ssf-alt</code>, <code>--ssf-maf</code>.)</p>
<table>
<thead>
<tr>
<th align="center">Flag</th>
<th align="center">--ssf-chr</th>
<th align="center">--ssf-snpid</th>
<th align="center">--ssf-ref</th>
<th align="center">--ssf-alt</th>
<th align="center">--ssf-maf</th>
<th align="center">--ssf-ss</th>
<th align="center">--ssf-beta</th>
<th align="center">--ssf-se</th>
<th align="center">--ssf-p</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">Default</td>
<td align="center">#CHROM</td>
<td align="center">ID</td>
<td align="center">REF</td>
<td align="center">A1</td>
<td align="center">A1_FREQ</td>
<td align="center">OBS_CT</td>
<td align="center">BETA</td>
<td align="center">SE</td>
<td align="center">P</td>
</tr>
</tbody>
</table>
<h3 id="2-phenotype-files">2) Phenotype Files</h3>
<p>Phenotype files can be provided to BridgePRS using the <code>--phenotype_files</code> flag. 
This must be a tab / space delimited file and missing data <strong>must</strong> be represented by either <code>NA</code> or <code>-9</code> (only for binary traits).
The first two column of the phenotype file should be the FID and the IID, and the rest can be phenotypes/covariates:  </p>
<table>
<thead>
<tr>
<th align="center">FID</th>
<th align="center">IID</th>
<th align="center">y</th>
<th align="center">y.binary</th>
<th align="center">PC1</th>
<th align="center">PC2</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">afr1_1</td>
<td align="center">afr2_1</td>
<td align="center">24.4</td>
<td align="center">1</td>
<td align="center">0.53</td>
<td align="center">0.950</td>
</tr>
<tr>
<td align="center">afr1_2</td>
<td align="center">afr2_2</td>
<td align="center">4.10</td>
<td align="center">0</td>
<td align="center">0.59</td>
<td align="center">0.450</td>
</tr>
<tr>
<td align="center">afr1_3</td>
<td align="center">afr2_3</td>
<td align="center">37.2</td>
<td align="center">1</td>
<td align="center">0.73</td>
<td align="center">-0.13</td>
</tr>
<tr>
<td align="center">afr1_4</td>
<td align="center">afr2_4</td>
<td align="center">5.40</td>
<td align="center">0</td>
<td align="center">0.44</td>
<td align="center">-0.55</td>
</tr>
</tbody>
</table>
<p>The phenotype of interest can be specified with the <code>--phenotype</code> flag and the covariates can be given as a comma separated list 
after the <code>--covariates</code> flag: </p>
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span>```
./bridgePRS check data -o out --pop AFR --phenotype y --covariates PC1,PC2 
                       --phenotype_files data/pop_africa/phenotypes/afr_test.dat data/pop_africa/phenotypes/afr_validation.dat
```
</pre></div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The column name(s) should not contain <em>space</em> nor <em>comma</em></p>
</div>
<h3 id="3-qc-snp-list">3) QC SNP List</h3>
<p>To select only SNPS that have passed QC, you can provide a file using the <code>--snp_file</code> flag. </p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../guide_requirements/" class="btn btn-neutral float-left" title="Requirements"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../guide_subprograms/" class="btn btn-neutral float-right" title="Subprograms">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/tadesouaiaia/BridgePRS-Guide" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../guide_requirements/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../guide_subprograms/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML" defer></script>
      <script src="../javascripts/details.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
